app/models/line_item.rb

```diff
@@ -25,7 +25,7 @@ class LineItem < ApplicationRecord
       where(["#{table_name}.created_at <= ? AND #{table_name}.finished_at >= ?", time, time])
   }
 
-  scope :oldest_first, -> { order(:created_at) }
+  scope :oldest_first, -> { unscope(:order).order(:id) }
```

Comment:
This is an example of Postgres timestamp and ordering issue. We took a screenshot of the line items of an invoice as they are rendered when ordered by `created_at` and `id` respectively. The values for these columns were added to table only for the purpose of the test, having `created_at` printed with milliseconds precision.

![screen shot 2019-02-18 at 7 54 32 pm](https://user-images.githubusercontent.com/1842261/53005192-04cc8700-3433-11e9-8853-a31384bb2937.png)

Reply:
Another solution would be to add a limit of precision in postgres but the inserts order are strange anyway.

The ID `1` is inserted after the ID `3` which is kinda counter-intuitive.